src/uct/ib/base/ib_md.c

```diff
@@ -1203,6 +1203,34 @@ static void uct_ib_md_check_dmabuf(uct_ib_md_t *md)
 #endif
 }
 
+int uct_ib_md_check_odp_common(uct_ib_md_t *md, const char **reason_ptr) {
+    if (IBV_ACCESS_ON_DEMAND == 0) {
+        *reason_ptr = "IBV_ACCESS_ON_DEMAND is not supported";
+        return 0;
+    }
+
+    if (!IBV_DEVICE_HAS_ODP(&md->dev)) {
+        *reason_ptr = "device does not support ODP";
+        return 0;
+    }
+
+    return 1;
+}
+
+void uct_ib_md_check_odp(uct_ib_md_t *md)
+{
+    const char *reason;
+
+    if (uct_ib_md_check_odp_common(md, &reason)) {
+        md->reg_nonblock_mem_types = UCS_BIT(UCS_MEMORY_TYPE_HOST);
+        ucs_debug("%s: ODP is supported, version 1",
+                  uct_ib_device_name(&md->dev));
+    } else {
+        ucs_debug("%s: ODP is disabled because %s",
+                  uct_ib_device_name(&md->dev), reason);
+    }
+}
```

Comment:
```suggestion
    if (!uct_ib_md_check_odp_common(md, &reason)) {
        ucs_debug("%s: ODP is disabled because %s",
                  uct_ib_device_name(&md->dev), reason);
        return;
    }

    md->reg_nonblock_mem_types = UCS_BIT(UCS_MEMORY_TYPE_HOST);
    ucs_debug("%s: ODP is supported, version 1",
              uct_ib_device_name(&md->dev));
}
```

Reply:
Done